src/ground_script_1.c

```diff
@@ -0,0 +1,1863 @@
+                    action->callbacks->func0C(action->parentObject, &pos);
+                    sub_80AC49C(res, &pos);
+                }
+                break;
+            }
+            case 0x1a: {
+                s8 unk;
+                GroundEffectData *eff;
+                Position32 pos;
+                s16 res;
+                s32 group;
+                s32 sector;
+                action->callbacks->func14(action->parentObject, &unk);
+                eff = ({ GroundEffectData eff = {
+                    .unk1 = unk,
+                    .widthTiles = 1,
+                    .heightTiles = 1,
+                    .pos = {},
+                    .kind = curCmd.arg2,
+                    .script = gFunctionScriptTable[curCmd.arg1].script,
+                    };
+                    group = curCmd.argShort < 0 ? scriptData->script.group : curCmd.argShort;
+                    sector = (s8)curCmd.argByte < 0 ? scriptData->script.sector : (s8)curCmd.argByte;
+                    &eff;
+                });
+                res = GroundEffect_Add(-1, eff, group, sector);
+                if (res >= 0) {
+                    action->callbacks->func0C(action->parentObject, &pos);
+                    sub_80AD3B4(res, &pos);
+                }
+                break;
+            }
+            case 0x1c: {
+                GroundMap_ExecuteEvent(curCmd.argShort, 1);
+                break;
+            }
+            case 0x1b: {
+                GroundMap_ExecuteEvent(curCmd.argShort, 0);
+                break;
+            }
+            case 0x1d: case 0x1e: {
+                s32 map;
+                s32 group, sector;
+                bool8 res;
+                {
+                    s32 tmp = curCmd.arg1 != -1 ? (s16)curCmd.arg1 : gCurrentMap;
+                    map = tmp;
+                }
+                {
+                    s32 tmp = curCmd.argShort < 0 ? scriptData->script.group : curCmd.argShort;
+                    group = tmp;
+                }
+                {
+                    s32 tmp = (s8)curCmd.argByte < 0 ? scriptData->script.sector : (s8)curCmd.argByte;
+                    sector = tmp;
+                }
+                map = sub_80A4D7C(map);
+                res = curCmd.op == 0x1e;
+                GroundMap_ExecuteStation(map, group, sector, res);
+                if (gUnknown_2039A34 != map) {
+                    gUnknown_2039A34 = map;
+                    sub_809D940();
+                    if (action->unk8[0] != 0)
+                        return 4; // Fatal?
+                }
+                break;
+            }
+            case 0x1f: {
+                s32 a = (s16)sub_8001658(0, 19);
+                const DungeonInfo *ret1 = sub_80A2608(a);
+                s32 thing = sub_8001784(0, 48, a) == 0 ? ret1->unk6 : ret1->unk8;
+                //s32 as = a;
+                //if (!(a >= 0x37 && a < 0x48) && (s16)sub_80A2750(a) == 1) {
+                // fakematch lmao
```

Comment:
Is this still a fakematch? If not then this commented-out code isn't needed anymore.

Reply:
I'm pretty sure this is still a fakematch, this is a range check like `0x37 <= a && a < 0x48` of some sort (which generates the correct code minus u32->u16 cast). I think an inline might be involved?